shared/chat/conversation/info-panel/add-to-channel/index.new.tsx

```diff
@@ -0,0 +1,201 @@
+              if (toAdd.has(item.username)) {
+                toAdd.delete(item.username)
+                setToAdd(new Set(toAdd))
+              } else {
+                toAdd.add(item.username)
+                setToAdd(new Set(toAdd))
+              }
+            }
+            return (
+              <Kb.ListItem2
+                firstItem={!Styles.isMobile || idx === 0}
+                icon={<Kb.Avatar size={32} username={item.username} />}
+                type="Small"
+                onClick={alreadyIn ? undefined : onCheck}
+                body={
+                  <Kb.Box2 direction="vertical" alignItems="flex-start">
+                    <Kb.ConnectedUsernames
+                      type="BodySemibold"
+                      colorFollowing={true}
+                      usernames={[item.username]}
+                    />
+                    <Kb.Text type="BodySmall" lineClamp={1}>
+                      {alreadyIn && <Kb.Text type="BodySmall">Already in{!!item.fullName && ' • '}</Kb.Text>}
+                      {item.fullName}
+                    </Kb.Text>
+                  </Kb.Box2>
+                }
+                action={
+                  <Kb.CheckCircle
+                    key={idx}
+                    onCheck={onCheck}
+                    checked={alreadyIn || toAdd.has(item.username)}
+                    disabled={alreadyIn}
+                    style={styles.checkCircle}
+                  />
+                }
+              />
+            )
+          }}
+          itemHeight={{sizeType: 'Small', type: 'fixedListItem2Auto'}}
+          style={styles.list}
+        />
+      </Kb.Box2>
+    </Kb.Modal>
+  )
+}
+
+const title = ({channelname, teamname}: {channelname: string; teamname: string}) =>
+  Styles.isMobile ? (
+    `Add to #${channelname}`
+  ) : (
+    <Kb.Box2 direction="vertical" gap="xtiny" alignItems="center" style={styles.title}>
+      {/* TODO move up 16 */}
+      <Kb.Avatar size={32} teamname={teamname} style={styles.avatar} />
+      <Kb.Box2 direction="vertical" alignItems="center">
+        <Kb.Text type="BodySmall" lineClamp={1}>
+          {teamname}
+        </Kb.Text>
+        <Kb.Text type="Header">Add to #{channelname}</Kb.Text>
+      </Kb.Box2>
+    </Kb.Box2>
+  )
+
+const styles = Styles.styleSheetCreate(() => ({
+  avatar: Styles.platformStyles({
+    isElectron: {
+      height: 16,
+      position: 'relative',
+      top: -16,
+    },
+  }),
+  checkCircle: {paddingRight: Styles.isMobile ? Styles.globalMargins.small : Styles.globalMargins.tiny},
+  filterInput: Styles.platformStyles({
+    isElectron: {
+      marginBottom: Styles.globalMargins.tiny,
+      marginLeft: Styles.globalMargins.small,
+      marginRight: Styles.globalMargins.small,
+      marginTop: Styles.globalMargins.tiny,
+    },
+  }),
+  list: Styles.platformStyles({isMobile: {height: '100%'}}),
+  listContainer: Styles.platformStyles({isElectron: {height: 370}}), // Maybe see if we could stop doing this
```

Comment:
Should this comment be a `TODO` and have a corresponding ticket?

Reply:
I'll change the comment - it was a shortcut because the list wouldn't expand the height of the modal.